static/app/components/modals/importDashboardFromFileModal.tsx

```diff
@@ -0,0 +1,127 @@
+import {Fragment, useState} from 'react';
+import {browserHistory} from 'react-router';
+import {css} from '@emotion/react';
+
+import {createDashboard} from 'sentry/actionCreators/dashboards';
+import {addErrorMessage, addSuccessMessage} from 'sentry/actionCreators/indicator';
+import {Button} from 'sentry/components/button';
+import {CodeSnippet} from 'sentry/components/codeSnippet';
+import {IconUpload} from 'sentry/icons';
+import {t} from 'sentry/locale';
+import {normalizeUrl} from 'sentry/utils/withDomainRequired';
+import {
+  assignDefaultLayout,
+  getInitialColumnDepths,
+} from 'sentry/views/dashboards/layoutUtils';
+import {Wrapper} from 'sentry/views/discover/table/quickContext/styles';
+
+function ImportDashboardFromFileModal({
+  Header,
+  Body,
+  Footer,
+  organization,
+  api,
+  location,
+}) {
+  const [dashboardData, setDashboardData] = useState('');
+  const [validated, setValidated] = useState(false);
+
+  function validateFile(fileToUpload) {
+    if (!(fileToUpload.type === 'application/json')) {
+      addErrorMessage('The uploaded file must be JSON');
+      setValidated(false);
+      return false;
+    }
+
+    setValidated(true);
+    return true;
+  }
+
+  const handleFileChange = e => {
+    if (e.target.files) {
```

Comment:
```suggestion
    if (e.target.files.length) {
```

If you add a file, then go back into the uploader but cancel instead of selecting another file, `e.target.files` will be an empty array which is truthy in JS and causes an error later when trying to validate the file